### ISSUE TITLE
Prevent accidental commits of generated `project` files

### ISSUE THREAD

**Issue by eecavanna** [microbiomedata/nmdc-schema#1602]
@turbomam wants files to exist in the `project` folder, but only the ones he explicitly puts there. 

While other people work in their local clones of the repo, updated versions of those files will be generated in their local environments. @turbomam does not want those updated versions to end up in the repo.

However, when @turbomam, himself, generates new, authoritative versions of those files, he does wants to be able to store them in the repo.

Related: #1567

**Comment by eecavanna** [comment]
I don't know that Git allows for this: store files in the repo, but ignore changes to those files. 🤷 Maybe this will involve a check performed by GitHub Actions.

**Comment by turbomam** [comment]
Thanks for thinking this though, @eecavanna 

Here are my goals: 
- all of the `project/`, `nmdc_schema/` and `examples/output` files generated by `make squeaky-clean all test` should be included in releases. I think @pkalita-lbl thinks including `examples/output/` and maybe some other generated files is excessive and I would be willing to reconsider. The `examples/output/` files are nice because they show the `src/data/valid/` files in their canonical forms. My release strategy has been
  1. merging pending PRs into `main`
  2. making and merging a new PR in which the version strings in `src/schema/nmdc.yaml` and `pyproject.toml` are updated
  3. checking out the latest `main`, running `make squeaky-clean all test`, and committing that directly into `main`. That could be modified if there is a better flow.
- generally speaking, PRs should not update any of those files, with one big exception: Python code in `nmdc_schema/` authored by an NMDC team member. This is in contrast with autogenerated Python files in that directory, like the following. I am open to re-organizing the authored code from the generated code, but that shoud to be done in a way that doesn't complicate building or using the package.
  - `nmdc.py`
  - `nmdc_schema_accepting_legacy_ids.py`
  - `nmdc_schema_merged.yaml`
 - it **would be nice** if changes to `project/`, authored code in `nmdc_schema/` and `examples/output` was not reported by `git status` and was blocked by the `.gitignore`, but if that isn't possible, we can write better documentation and/or restructure `nmdc_schema/`
- if merging the `berkeley-schema-fy24` fork into `nmdc-schema` will cause conflicts in `project/`, `nmdc_schema/` or `examples/output`, we could make an exception an allow that, following the release flow I have documented above

PS: when @aclum's Napa re-iding squad is done, we can modify the Makefiles so that the `nmdc_schema_accepting_legacy_ids*` files aren't generated any more.

**Comment by turbomam** [comment]
## Status update (2026-02-27)

The situation has evolved since the original discussion:

### What's changed
- **PR #2302** (merged 2026-02-21) added a [Downloads page](https://microbiomedata.github.io/nmdc-schema/downloads/) to the docs website, providing an alternative distribution channel for generated schema artifacts. This closed #1960.
- **`project/` directory** is now fully gitignored and no longer committed.
- **`src/schema/mixs.yaml`** is generated via a dedicated pipeline (`make src/schema/mixs.yaml`) and has its own clean target (`make mixs-yaml-clean`).
- **Makefile reorganization** (PR #2848) split targets across `makefiles/mixs.Makefile` and `makefiles/migrators.Makefile`.

### What hasn't changed
The following generated files are **still tracked in git** and show up as modified after every `make squeaky-clean all test`:
- `nmdc_schema/nmdc.py`
- `nmdc_schema/nmdc.schema.json`
- `nmdc_schema/nmdc_materialized_patterns.json`
- `nmdc_schema/nmdc_materialized_patterns.schema.json`
- `nmdc_schema/nmdc_materialized_patterns.yaml`
- `nmdc_schema/nmdc_pydantic.py`
- `nmdc_schema/gold-to-mixs.sssom.tsv`

Contributors routinely have to avoid committing these, and they still cause noise in `git status` and PR diffs.

### Remaining blocker
Downstream consumers (nmdc-runtime, submission-schema) fetch some of these files via **raw GitHub URLs at tagged commits**. Until those consumers switch to PyPI or the Downloads page, the files must exist at the tagged commit. Past incidents (#2150, #2176, #2300) demonstrate what happens when they're missing at release time.

### Proposed path forward
1. **Audit downstream raw-URL consumers** — identify exactly which repos/files depend on raw GitHub access to these artifacts
2. **Migrate consumers** to PyPI package imports or the Downloads page
3. **Add a CI release workflow** that generates and commits these files only at release time (tag-triggered)
4. **Gitignore them for day-to-day development** once consumers no longer need them at arbitrary commits

Sierra Moxon also advocated for this direction (Jul 2025): _"Use .gitignore wisely, leaving little chance for users to accidentally commit generated files."_

See also: #1230, #1960
